File: Competitive Programming 3/Section 1.4/Palidromes/10018 - Reverse and Add/Python/main.py
from sys import stdin

def isPalindrome(s):
    return s == s[::-1]
def reverseAndAdd(s):
    return str(int(s)+int(s[::-1]))

def main():
    input()
    for s in stdin:
        ite = 0
        while(not isPalindrome(s)):
            s = reverseAndAdd(s)
            ite+=1
        print(ite,s)
main()

File: Competitive Programming 3/Section 1.4/Palidromes/10018 - Reverse and Add/C++/main.cpp
#include<bits/stdc++.h>

using namespace std;

bool isPalindrome(const string &s){
    int lim = s.size() >> 1;
    for(int i=0;i<lim;i++)
        if(s[i]!=s[s.size()-1-i])
            return false;
    return true;
}
string reverseAndAdd(const string &s){
    string sR = s;
    reverse(sR.begin(),sR.end());
    return to_string(stoul(s)+stoul(sR));
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int n;
    cin >> n;
    string s;
    while(n--){
        cin >> s;
        int ite=0;
        for(;!isPalindrome(s);ite++)
            s = reverseAndAdd(s);
        cout << ite << " " << s << "\n";
    }
    return 0;
}